src/execution_plan/optimizations/traverse_order.c

```diff
@@ -17,51 +17,50 @@
 typedef AlgebraicExpression **Arrangement;
 
 // Create a new arrangement.
-Arrangement Arrangement_New(uint size) {
+static inline Arrangement Arrangement_New(uint size) {
 	return rm_malloc(sizeof(AlgebraicExpression *) * size);
 }
 
 // Clone arrangement.
-Arrangement Arrangement_Clone(const Arrangement arrangement, uint size) {
+static inline Arrangement Arrangement_Clone(const Arrangement arrangement, uint size) {
```

Comment:
I think it probably doesn't matter; at O3 the compiler may choose to inline functions that aren't explicitly `inline`, and it may always choose to not inline functions that are. The stylistic concept I've been following recently is that if a function is non-recursive, short, and simple (loop, lookup, allocation), it may as well be inlined. Do you think that's an overkill approach?

Reply:
Incase the function is used in lots of places then yes, due to code size being enlarged,
I tend to think of inline as a way to improve performance when dealing with critical code paths